internal/operations/operations.go

```diff
@@ -199,6 +199,47 @@ func Ping(ctx context.Context, dut binding.DUT, dest string, count int32) error
 	return nil
 }
 
+// GetSystemTime returns the current time in nanoseconds From epoch
+func GetSystemTime(ctx context.Context, dut binding.DUT) (uint64, error) {
+	gnoi, err := FetchGNOI(ctx, dut)
+	if err != nil {
+		return 0, err
+	}
+	resp, err := gnoi.System().Time(ctx, &spb.TimeRequest{})
+
+	if err != nil {
+		return 0, err
+	}
+	return resp.GetTime(), nil
+
+}
+
+// FactoryReset executes the Factory Reset RPC
+func FactoryReset(ctx context.Context, dut binding.DUT, factoryOs bool, zeroFill bool) error {
+
+	gnoi, err := FetchGNOI(ctx, dut)
+	if err != nil {
+		return err
+	}
+	resp, err := gnoi.FactoryReset().Start(ctx, &frpb.StartRequest{
+		FactoryOs: factoryOs,
+		ZeroFill:  zeroFill,
+	})
+	if err != nil {
+		return err
+	}
+	log.Infof("Factory Reset Response: %v", resp)
+	rpcErr := resp.GetResetError()
+	if rpcErr != nil {
+		return fmt.Errorf("Error Response From Factory Reset Operation, %v ", rpcErr)
+	}
+	if resp.GetResetSuccess() == nil {
```

Comment:
I think you can drop this whole if-statement. If there is no error, it succeeded